x/concentrated-liquidity/internal/math/math.go

```diff
@@ -96,22 +96,15 @@ func CalcAmount1Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec, roundUp bool) sdk.Dec
 
 // getNextSqrtPriceFromAmount0RoundingUp utilizes the current squareRootPrice, liquidity of denom0, and amount of denom0 that still needs
 // to be swapped in order to determine the next squareRootPrice
-// TODO: make an issue to determine if we can remove the less precise formula
+// Note: we are using only using the precise formula here.
 func GetNextSqrtPriceFromAmount0RoundingUp(sqrtPriceCurrent, liquidity, amountRemaining sdk.Dec) (sqrtPriceNext sdk.Dec) {
 	if amountRemaining.Equal(sdk.ZeroDec()) {
 		return sqrtPriceCurrent
 	}
 
 	product := amountRemaining.Mul(sqrtPriceCurrent)
-	// use precise formula if product doesn't overflow
-	if (product.Quo(amountRemaining)).Equal(sqrtPriceCurrent) {
-		denominator := liquidity.Add(product)
-		if denominator.GTE(liquidity) {
-			return liquidity.Mul(sqrtPriceCurrent).QuoRoundUp(denominator)
-		}
-	}
-	// if the product does overflow, use less precise formula
-	return liquidity.QuoRoundUp(liquidity.Quo(sqrtPriceCurrent).Add(amountRemaining))
+	denominator := liquidity.Add(product)
+	return liquidity.Mul(sqrtPriceCurrent).QuoRoundUp(denominator)
```

Comment:
Were we never hitting the if statement where we use the less precise formula in our test cases? Seems odd to me that we remove the less precise formula but none of the tests are breaking

Reply:
No, we never hit it. Also, it really isn't "less precise", it is essentially the same formula but just transformed in a different way. My guess was it was needed for order of operations requirements, but I am still not completely sure